### PR TITLE
first attempt to export html

### PR DESCRIPTION
**Pull request by mslourens**
minimum viable implementation of #1233, please let me know what you think.

### CODE REVIEW

**Comment by kazup01** [comment]
Awesome❗
I think that it would be better if there is feature that share notes on the web, what do you think? @mslourens

**Comment by mslourens** [comment]
What do you mean by 'share notes on the web'? The feature request was about
exporting notes to HTML (including images and other references), so what
would this export to the web look like? Where on the web would you export
it?

Groeten, Maurits

On Dec 9, 2017 06:21, "Kazz Yokomizo" <notifications@github.com> wrote:

Awesome❗
I think that it would be better if there is feature that share notes on the
web, what do you think? @mslourens <https://github.com/mslourens>

—
You are receiving this because you were mentioned.

Reply to this email directly, view it on GitHub
<https://github.com/BoostIO/Boostnote/pull/1256#issuecomment-350425132>,
or mute
the thread

**Comment by kazup01** [comment]
I understand this pull request. I think that it would be better if could display the notes on web page as a future plan.

File: browser/components/MarkdownPreview.js [hunk 1]
```diff
@@ -173,21 +174,29 @@ export default class MarkdownPreview extends React.Component {
     this.exportAsDocument('md')
   }
 
+  handleSaveAsHtml () {
+    this.exportAsDocument('html', function formatHtml (value) {
```
Comment (sota1235) [inline browser/components/MarkdownPreview.js]: you don't need to name function. plz just write such as `(value) => this.refs.root.contentWindow.document.documentElement.outerHTML`

File: browser/components/MarkdownPreview.js [hunk 2]
```diff
@@ -173,21 +174,29 @@ export default class MarkdownPreview extends React.Component {
...
 
-  exportAsDocument (fileType) {
+  exportAsDocument (fileType, formatter) {
     const options = {
       filters: [
         { name: 'Documents', extensions: [fileType] }
       ],
       properties: ['openFile', 'createDirectory']
     }
+    var value = formatter ? formatter.call(this, this.props.value) : this.props.value
```
Comment (sota1235) [inline browser/components/MarkdownPreview.js]: plz use `const`

File: lib/main-menu.js [hunk 3]
```diff
@@ -106,6 +106,13 @@ const file = {
             mainWindow.webContents.send('list:isMarkdownNote')
             mainWindow.webContents.send('export:save-md')
           }
+        },
+        {
+          label: 'Html (.html)',
```
Comment (sota1235) [inline lib/main-menu.js]: `HTML` is better, I think

**Comment by kazup01** [comment]
Merged. Thanks for your contribution @mslourens !

**Pull request merged**